### DISCUSSION TITLE
Proposal to add support for space-separated migration names in the sea-orm-cli tool.

### DISCUSSION THREAD

**Discussion by babalolajnr** [SeaQL/sea-orm#1536] [Ideas]
Currently, the sea-orm-cli tool only supports generating migrations using underscore-separated names. For example, the following command generates a migration named "create_user_table":
```bash
$ sea-orm-cli migrate generate create_user_table
```
While this works well, it can be cumbersome to type out long migration names that require underscores after each word. To address this, I propose adding support for space-separated migration names in the sea-orm-cli tool. This would allow users to generate migrations using a more natural and convenient syntax. For example, the following command would generate a migration named "create user table":
```bash
$ sea-orm-cli migrate generate "create user table"
```
This change would improve the user experience of the sea-orm-cli tool, making it easier and more intuitive to use.

**Technical Implementation**
To implement this feature, I propose adding a new command-line option to the sea-orm-cli migrate generate command. This option would allow users to specify a space-separated migration name. The tool would then convert the space-separated name to an underscore-separated name before generating the migration file.
Specifically, the implementation would involve the following steps:

1. Parse the command-line arguments to extract the migration name.
2. Convert the migration name to an underscore-separated format by replacing spaces with underscores.
3. Generate the migration file using the converted migration name.

To ensure that the tool works as expected, thorough testing would be required. This would involve testing the tool with various migration names, including ones with spaces and underscores, to verify that the conversion from space-separated to underscore-separated names works correctly.

**Conclusion**
In summary, adding support for space-separated migration names in the sea-orm-cli tool would improve the usability and convenience of the tool for users. This proposal outlines a straightforward technical implementation for this feature, and thorough testing would be required to ensure its reliability. As a first-time contributor, I am excited about the opportunity to work on this feature and contribute to the sea-orm-cli project. I would also like to stick around, use this as an opportunity to get my feet wet with the codebase.

**Comment by billy1624** [comment]
Hey @babalolajnr, interesting!! Thanks for the proposal! I think this is something that worth doing :)

**Reply by babalolajnr** [reply]
I will try to work on this. Thanks for the response

**Reply by babalolajnr** [reply]
A [PR](https://github.com/SeaQL/sea-orm/pull/1570) has been created for this.
